Add debloat list for Tecno smartphones

Mayank-9018 opened the pull request:
Adds debloat list for Tecno Smartphones. Will also work on other manufacturers like infinix.

0x192 commented on resources/assets/uad_lists.json:
```diff
@@ -18925,5 +18925,196 @@
+    "labels": null,
+    "removal": "Recommended"
+  },
+  {
+    "id": "com.transsion.filemanagerx",
+    "list": "Oem",
+    "description": "File Manager App.",
+    "neededBy": null,
+    "labels": null,
+    "removal": "Recommended"
+  },
+  {
+    "id": "com.gallery20",
+    "list": "Oem",
+    "description": "AI Gallery. Gallery app.",
+    "neededBy": null,
+    "labels": null,
+    "removal": "Recommended"
+  },
```
I don't think that's a good idea to put the default File Manager and Gallery apps in the `Recommended` list, even if those apps are really bad (which I bet they are)

I'd rather put them in the `Advanded` list.

0x192 replied:
So there are by default 2 _File Manager_ apps and 2 _Gallery_ apps on Tecno Phones? Those of the manufacturer and those of Google.

The question is: can we assume it is the case for all the Tecno phones? I'm not sure about that.

0x192 replied:
I improved some descriptions and moved `com.transsion.filemanagerx` and `com.gallery20` to the `Advanced` list.

Your phone has really awful pre-installed apps that's incredible!

0x192 commented:
Thanks for your contribution! The rest looks good to me :)

0x192 commented:
There are some apps that cannot be found on the Web. Would you mind uploading them?

I wrote a small Bash script to help you pull the apks. Are you on Linux/MacOS?

<details>
<summary>Show script</summary>

```bash
#!/usr/bin/env bash

set -Eeuo pipefail

declare -ra PACKAGES=(
    "com.transsion.hiparty"
    "com.talpa.share"
    "net.bat.store"
    "com.transsnet.store"
    "com.transsion.plat.appupdate"
    "com.idea.questionnare"
    "com.gallery20"
    "com.transsion.videocallenhancer"
    "com.transsion.kolun.assistant"
    "com.transsion.health"
    "com.transsion.phonemanager"
    "com.transsion.smartpanel")

mkdir -p uad_apks && cd uad_apks

for p in "${PACKAGES[@]}"; do
    adb pull "$(adb shell pm path "$p" | sed 's/package://g')"
done
```

</details>

If you are on Windows, I'll give you a Powershell script.

You can then archive the APKs in a zip file and upload it on [send.tresorit.com/](https://send.tresorit.com/) and I'll upload them to [ApkMirror](https://www.apkmirror.com/) and [Pithus](https://beta.pithus.org/).

Or you can directly upload yourself the apks on APKMirror and Pithus :)

Mayank-9018 commented:
I'll upload them probably by the end of this week.

0x192 commented:
@Mayank-9018 ? :slightly_smiling_face:

Mayank-9018 commented:
Thanks for reminding. Forgot about it. Started upload to APKMirror. Have to update the `uad_lists.json` too. What should I do after upload completes?

0x192 commented:
Nothing special. I'll analyze the APK with [Pithus](https://beta.pithus.org/) and update the descriptions.

Thanks for answering this fast!

Mayank-9018 commented:
For some reason, upload on APKMirror would keep failing. So, I uploaded them on tresoritsend here: https://send.tresorit.com/a#V-40WmCFzGBKhZcu4ckR_w

0x192 commented:
I finally had the time to decompile the APKs you sent me. I found a very questionable piece of code in `com.transsion.hamal` (TranHamal).

```java
        if (!GetInfoUtils.judgeWhiteUser()) {
            Tlog.m4d("Is Black Owner,Can't Run Hamal!");
            SPUtils.setStringValue("last_finish_type", "Is Black Owner,Can't Run Hamal", false);
            SPUtils.setStringValue("last_run_spend_time_value", "0s", true);
            if (!TextUtils.isEmpty(SystemProperties.get("persist.sys.transsion.hamalwhiteowner", (String) null))) {
                Tlog.m4d("Hamal service Is Black Owner,Can't Run Hamal!");
                stopSelf();
            } else {
                Tlog.m4d("Hamal service download config!");
                GetInfoUtils.getRdTimeZoneConfig();
            }
            return 2;
        }
...
if (1 == SystemProperties.getInt("ro.vendor.transsion.aftersales.support", 0)) {
            if ("afterSales".equals(intent != null ? intent.getStringExtra("type") : "")) {
                Tlog.m4d("afterSales start");
                if (new File("/tranfs/SDD/exception").exists()) {
                    new Thread(new Runnable(this) {
                        @Override // java.lang.Runnable
                        public void run() {
                            try {
                                NetworkUtils.tarAfterSales();
                            } catch (IOException e) {
                                e.printStackTrace();
                            }
                            try {
                                ActivityManager.getService().startTNE("0x007a0015", 0L, 0, "/tranfs/SDD/exception.zip");
                            } catch (Exception e2) {
                                Tlog.m1e("Can't call ActivityManager.startTNE", e2);
                            }
                        }
                    }).start();
                } else {
                    Tlog.m4d("DIR /tranfs/SDD/exceptiondo not exists.");

```
I saw you didn't add this app in the `uad_lists.json` file. It can send stuff on the Microsoft Azure Cloud. Do you know the purpose of this app?

The app also check if the phone has an India country code and use a special Azure account in this case.

You may want to have a look at `/data/user/0/com.transsion.tranfansfeedback/files/upload_wholeLog` and see if stuff have been uploaded.

What does `Hamal` means in this context? And what is this `aftersales` settings? Something related to the phone warranty?

I hope you can help me on this :slightly_smiling_face: 

FYI : https://beta.pithus.org/report/35fd79ebbe51808196605146a62aaef13bc654477d917078a3ae5d3f06ba8836

Mayank-9018 commented:
> . I found a very questionable piece of code in `com.transsion.hamal` (TranHamal).

That doesn't look good.

> I saw you didn't add this app in the `uad_lists.json` file. It can send stuff on the Microsoft Azure Cloud. Do you know the purpose of this app?

Just added. No idea what it does though.

> The app also check if the phone has an India country code and use a special Azure account in this case.

It's spooky.

> You may want to have a look at `/data/user/0/com.transsion.tranfansfeedback/files/upload_wholeLog` and see if stuff have been uploaded.

I don't have that folder, I removed the apps soon as I factory reset.
 
> What does `Hamal` means in this context? And what is this `aftersales` settings? Something related to the phone warranty?

No idea what 'hamal' means. You're probably right after sales would be related to phone warranty and other after-sales services like chat and whatnot.

0x192 commented:
Too much conflicts with the main branch. I rebased all the changes in #285

It was a long ride! Thanks for your help! :slightly_smiling_face:
